app/routes/dashboard.tsx

```diff
@@ -1,73 +1,30 @@
-import { useLoaderData, useNavigate } from "@remix-run/react";
+import HankoStarterInfo from "../hanko starter components/HankoStarterInfo";
+import HankoStarterDashboard from "../hanko starter components/HankoStarterDashboard";
+import HankoStarterHeader from "../hanko starter components/HankoStarterHeader";
 
-import {
-    Suspense,
-    useEffect,
-    useState,
-} from "react";
-import {
-    type Hanko,
-    register,
-} from "~/utils/hanko.client";
 
+import { type LoaderFunction, redirect } from "@remix-run/node";
+import { ValidateCurrentSession } from "../services/auth.server";
 
-export const loader = () => {
-    return { hankoUrl: process.env.HANKO_API_URL };
-};
-
-function HankoProfile() {
-    let data = useLoaderData<typeof loader>();
-    let hankoUrl = data.hankoUrl || '';
-
-    const [hanko, setHanko] = useState<Hanko>();
-    const navigate = useNavigate();
-
-
-    useEffect(() => {
-        register(hankoUrl)
-            .catch((error: Error) => {
-                console.error(error.message);
-            })
-            .then((result) => {
-                if (result) {
-                    setHanko(result.hanko);
-                }
-            });
-    }, [hankoUrl]);
+export const loader: LoaderFunction = async ({request}) => {
+    const hankoUrl = process.env.HANKO_API_URL || "";
+    const validated = await ValidateCurrentSession(request , hankoUrl);
 
+    if(!validated){
+      return redirect("/");//Path to redirect to if user is not authenticated
+    }
 
+    return { hankoUrl: hankoUrl }; 
+};
 
-    const logout = async () => {
-        try {
-            await hanko?.user.logout();
-            navigate("/login");
-            return;
-        } catch (error) {
-            console.error("Error during logout:", error);
-        }
-    };
-
-    return (
-        <div className="">
-            <Suspense fallback={"Loading..."}>
-                <hanko-profile />
-            </Suspense>
-            <div>
-                <button onClick={logout}>Logout</button>
-                <button onClick={() => navigate("/protected")}>Access Protected page</button>
-            </div>
-        </div>
-    )
-}
-
-
-const Dashboard = () => {
 
-    return (
-        <div>
-            <HankoProfile />
-        </div>
-    )
+const dashboardPage = () => {
```

Comment:
Please use uppercase for the first letter, as the other components.
```suggestion
const DashboardPage = () => {
```